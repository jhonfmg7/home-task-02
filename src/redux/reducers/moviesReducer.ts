import Movie from "../../types/movie.interface";
import { GET_MOVIES_ERROR, GET_MOVIES_START, GET_MOVIES_SUCCESS } from "../types";

interface InitialState {
    movies: Movie[],
    moviesQuantity: number,
    loading: boolean,
    error: Error | any
}

interface Payload {
    movies: Movie[],
    quantity: number
}

interface Action {
    type: string,
    payload?: Payload
}

const initialState: InitialState = {
    movies: [],
    moviesQuantity: 0,
    loading: false,
    error: null
}

export default (state = initialState, action: Action) => {
    switch (action.type) {
        case GET_MOVIES_START:
            return {
                ...state,
                loading: true,
                error: null,
                movies: []
            }
        case GET_MOVIES_SUCCESS:
            return {
                ...state,
                loading: false,
                error: null,
                movies: action.payload.movies,
                moviesQuantity: action.payload.quantity
            }
        case GET_MOVIES_ERROR:
            return {
                ...state,
                loading: false,
                error: action.payload,
                movies: []
            }
        default:
            return state;
    }
}