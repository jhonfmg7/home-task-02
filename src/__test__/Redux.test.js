import moviesReducer from "../redux/reducers/moviesReducer";

test('<Main /> validate redux reducer when you pass a empty type', () => {
    const initialState = undefined;
    const action = { type: "" }
    const result = moviesReducer(initialState, action);
    
    expect( Object.keys(result).length ).toBeGreaterThan(0)
    expect( result ).toMatchObject({
        deleteLoading: false,
        error: null,
        errorDelete: null,
        isOpenMoreInfoModal: false,
        loading: false,
        message: null,
        movieSelected: null,
        movies: [],
        moviesQuantity: 0,
        reload: true,
        sortBySelected: "release_date",
        typeSelected: "all",
    });
});