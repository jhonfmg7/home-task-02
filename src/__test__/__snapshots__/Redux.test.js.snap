// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> validate redux reducer when you pass a empty type 1`] = `
{
  "deleteLoading": false,
  "error": null,
  "errorDelete": null,
  "isOpenMoreInfoModal": false,
  "loading": false,
  "message": null,
  "movieSelected": null,
  "movies": [],
  "moviesQuantity": 0,
  "reload": true,
  "sortBySelected": "release_date",
  "typeSelected": "all",
}
`;
